packages/trpc/server/routers/viewer/workflows/util.ts

```diff
@@ -99,16 +95,80 @@ export const verifyEmailSender = async (email: string, userId: number, teamId: n
           teamId,
         },
       });
-    } else if (verifiedEmail.teamId !== teamId) {
+    } else if (teamId && verifiedEmail.teamId !== teamId) {
+      await prisma.verifiedEmail.create({
+        data: {
+          email,
+          userId,
+          teamId,
+        },
+      });
+    }
+    return;
```

Comment:
nit: I think earlier approach of a container if conditition with `teamId` was better instead of teamId being checked with both if and else if.

It makes more sense logically.